### DISCUSSION TITLE
Do CC threads run inside the normal game loop?

### DISCUSSION THREAD

**Discussion by FireIP** [cc-tweaked/CC-Tweaked#1819] [Q&A] [answered]
Do CC threads run inside the normal game loop? Or are they run in parallel?

**Comment by SquidDev** [comment, marked as answer]
All Lua code is executed on a separate thread(s)[^1], as this means misbehaving computers do not stall the main game loop. Some logic is executed on the main thread (for instance, peripheral calls), but these are normally quite fast.

[^1]: The number of threads can be configured with the `computer_threads` config option.

**Reply by FireIP** [reply]
Thanks! This is great news! As the "singlethreaded" nature of Minecraft  means I have ~6 cores idle at all times and can give those to CC.
